test/spec/unit/core/targeting_spec.js

```diff
@@ -326,4 +326,46 @@ describe('targeting tests', function () {
       });
     });
   });
+
+  describe('setTargetingForAst', function () {
+    let sandbox;
+    beforeEach(function() {
+      sandbox = sinon.createSandbox();
+      sandbox.stub(targetingInstance, 'resetPresetTargetingAST').callsFake(function() {});
+      sandbox.stub(window.apntag, 'setKeywords').callsFake(function() {});
+    });
+    afterEach(function () {
+      sandbox.restore();
+    });
+
+    function mockGetAllTargeting(adUnitCodes) {
+      adUnitCodes = Array.isArray(adUnitCodes) ? adUnitCodes : Array(adUnitCodes);
+      sandbox.stub(targetingInstance, 'getAllTargeting').callsFake(function() {
```

Comment:
I would suggest here to use `stub.returns`. Changing the implementation of getAllTargeting with this fake function is not a good idea. So you can do something like below in each test case
```
targetingStub.returns({
   'adUnitCode': {hb_bidder:'appnexus'}
});
```

Reply:
Done.